app/scripts/models/auth_brokers/web-channel.js

```diff
@@ -96,14 +96,43 @@ define([
                 this, account, { closeWindow: true });
     },
 
+    beforeSignUpConfirmationPoll: function (account) {
+      // If the relier wants keys, the signup verification tab will need
+      // to be able to fetch them in order to complete the flow.
+      // Send them as part of the oauth session data.
+      if (this.relier.wantsKeys()) {
```

Comment:
Maybe this should be done in `persist` so this same logic can be used for account unlock and password reset as well?